csm/conf/configure.py

```diff
@@ -156,16 +153,16 @@ def _set_csm_endpoint():
 
     @staticmethod
     def _set_s3_endpoints():
-        s3_endpoints = Conf.get(const.CONSUMER_INDEX, const.RGW_S3_DATA_ENDPOINTS_KEY)
-        if s3_endpoints:
-            Log.info(f"Fetching s3 endpoint.{s3_endpoints}")
-            s3_endpoints_count = len(s3_endpoints)
-            for endpoint_count in range(s3_endpoints_count):
+        s3_num_eps = Conf.get(const.CONSUMER_INDEX,
+            const.RGW_NUM_ENDPOINTS_KEY)
+        for ep_count in range(int(s3_num_eps)):
+            ep = Conf.get(const.CONSUMER_INDEX,
+                f'{const.RGW_S3_DATA_ENDPOINTS_KEY}[{ep_count}]')
+            if ep:
```

Comment:
You are raising an error just by checking the 0th index. What if your list is like - 
["", "Endpoint 1", "Endpoint 2"]

Reply:
@pawankumarsrivastava we are iterating over num_endpoints key present in confstore, whatever value is present there, these many valid values should be there in the list, that is why checking each value if not found then raising the error.

Reply:
made changes as per our discussion. done.